illumina_uploader/mcms.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python
+
+from datetime import datetime, timedelta
+import json
+from requests import post
+from requests.auth import HTTPBasicAuth
+from time import sleep
+
+
+class McmsEmailService:
+
+    def __init__(self, auth_url, client_id, client_secret, sender_email, logger):
+        self.auth_url = auth_url
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.sender_email = sender_email
+        self.logger = logger
+        self.token_obj = None
+
+    def _post_email(self, email_url, email_request_body):
+        """
+        Sends email HTTP POST request.
+
+        :return: If POST was successful
+        :rtype: boolean
+        """
+        headers = {
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+            "Authorization": "Bearer " + self.token_obj["access_token"],
+        }
+        response = post(
+            email_url, data=json.dumps(email_request_body), headers=headers
+        )
+        if response.status_code == 200:
+            self.logger.info(
+                f'Successfully sent email: {datetime.now().isoformat()}\n{json.dumps(email_request_body["email"])}'
+            )
+            return True
+        else:
+            self.logger.error(
+                f'Failed to send email\nstatus:{response.status_code} - {response.text}\n{json.dumps(email_request_body["email"])}'
+            )
+            return False
+
+    def _fetch_token(self):
```

Comment:
using '_' to indicate private method. Users of McmsEmailService should really only be using send_email()